diff --git a/node_modules/@polkadot/types-create/packageInfo.js b/node_modules/@polkadot/types-create/packageInfo.js
index fb44350..399f19c 100644
--- a/node_modules/@polkadot/types-create/packageInfo.js
+++ b/node_modules/@polkadot/types-create/packageInfo.js
@@ -1 +1 @@
-export const packageInfo = { name: '@polkadot/types-create', path: (import.meta && import.meta.url) ? new URL(import.meta.url).pathname.substring(0, new URL(import.meta.url).pathname.lastIndexOf('/') + 1) : 'auto', type: 'esm', version: '16.5.4' };
+export const packageInfo = { name: '@polkadot/types-create', path: 'auto', type: 'cjs', version: '16.5.4' };
