diff --git a/node_modules/@polkadot/ui-shared/cjs/icons/polkadot.js b/node_modules/@polkadot/ui-shared/cjs/icons/polkadot.js
index 390c797..42b52fe 100644
--- a/node_modules/@polkadot/ui-shared/cjs/icons/polkadot.js
+++ b/node_modules/@polkadot/ui-shared/cjs/icons/polkadot.js
@@ -55,7 +55,7 @@ const SCHEMA = {
 const OUTER_CIRCLE = {
   cx: C,
   cy: C,
-  fill: '#eee',
+  fill: 'transparent',
   r: C
 };
 let zeroHash = new Uint8Array();
diff --git a/node_modules/@polkadot/ui-shared/icons/polkadot.js b/node_modules/@polkadot/ui-shared/icons/polkadot.js
index 8b12e00..b72c5e2 100644
--- a/node_modules/@polkadot/ui-shared/icons/polkadot.js
+++ b/node_modules/@polkadot/ui-shared/icons/polkadot.js
@@ -49,7 +49,7 @@ const SCHEMA = {
 const OUTER_CIRCLE = {
   cx: C,
   cy: C,
-  fill: '#eee',
+  fill: 'transparent',
   r: C
 };
 let zeroHash = new Uint8Array();
