diff --git a/node_modules/@polkadot/util/cjs/versionDetect.js b/node_modules/@polkadot/util/cjs/versionDetect.js
index af482b7..d83218b 100644
--- a/node_modules/@polkadot/util/cjs/versionDetect.js
+++ b/node_modules/@polkadot/util/cjs/versionDetect.js
@@ -75,7 +75,8 @@ function warn(pre, all, fmt) {
  * @description A `@polkadot/*` version detection utility, checking for one occurrence of a package in addition to checking for dependency versions.
  */
 function detectPackage({ name, path, type, version }, pathOrFn, deps = []) {
-    if (!name.startsWith('@polkadot')) {
+    console.log('name', name);
+    if (!name.startsWith('@polkadot') && !name.startsWith('@subwallet')) {
         throw new Error(`Invalid package descriptor ${name}`);
     }
     const entry = getEntry(name);
diff --git a/node_modules/@polkadot/util/packageInfo.js b/node_modules/@polkadot/util/packageInfo.js
index f146eee..6ecf83a 100644
--- a/node_modules/@polkadot/util/packageInfo.js
+++ b/node_modules/@polkadot/util/packageInfo.js
@@ -1 +1,6 @@
-export const packageInfo = { name: '@polkadot/util', path: (import.meta && import.meta.url) ? new URL(import.meta.url).pathname.substring(0, new URL(import.meta.url).pathname.lastIndexOf('/') + 1) : 'auto', type: 'esm', version: '12.6.2' };
+export const packageInfo = {
+    name: '@polkadot/util',
+    path: 'auto',
+    type: 'cjs',
+    version: '12.6.2'
+};
