diff --git a/node_modules/@polkadot/types/packageInfo.js b/node_modules/@polkadot/types/packageInfo.js
index f6fd584..35c35bf 100644
--- a/node_modules/@polkadot/types/packageInfo.js
+++ b/node_modules/@polkadot/types/packageInfo.js
@@ -1 +1 @@
-export const packageInfo = { name: '@polkadot/types', path: (import.meta && import.meta.url) ? new URL(import.meta.url).pathname.substring(0, new URL(import.meta.url).pathname.lastIndexOf('/') + 1) : 'auto', type: 'esm', version: '16.4.2' };
+export const packageInfo = { name: '@polkadot/types', path: 'auto', type: 'cjs', version: '16.4.2' };
