diff --git a/node_modules/@polkadot/reactnative-identicon/packageInfo.js b/node_modules/@polkadot/reactnative-identicon/packageInfo.js
index 245a8b5..fdd749f 100644
--- a/node_modules/@polkadot/reactnative-identicon/packageInfo.js
+++ b/node_modules/@polkadot/reactnative-identicon/packageInfo.js
@@ -1 +1 @@
-export const packageInfo = { name: '@polkadot/reactnative-identicon', path: (import.meta && import.meta.url) ? new URL(import.meta.url).pathname.substring(0, new URL(import.meta.url).pathname.lastIndexOf('/') + 1) : 'auto', type: 'esm', version: '3.16.4' };
+export const packageInfo = { name: '@polkadot/reactnative-identicon', path: 'auto', type: 'cjs', version: '3.16.4' };
