diff --git a/node_modules/@subwallet/extension-base/services/swap-service/utils.js b/node_modules/@subwallet/extension-base/services/swap-service/utils.js
index 1b57d06..e4132af 100644
--- a/node_modules/@subwallet/extension-base/services/swap-service/utils.js
+++ b/node_modules/@subwallet/extension-base/services/swap-service/utils.js
@@ -1,7 +1,6 @@
 // Copyright 2019-2022 @subwallet/extension-base
 // SPDX-License-Identifier: Apache-2.0
 
-import { Assets, Chains } from '@chainflip/sdk/swap';
 import { COMMON_ASSETS, COMMON_CHAIN_SLUGS } from '@subwallet/chain-list';
 import { _getAssetDecimals } from '@subwallet/extension-base/services/chain-service/utils';
 import { SwapProviderId } from '@subwallet/extension-base/types/swap';
@@ -9,22 +8,22 @@ import BigN from 'bignumber.js';
 export const CHAIN_FLIP_TESTNET_EXPLORER = 'https://blocks-perseverance.chainflip.io';
 export const CHAIN_FLIP_MAINNET_EXPLORER = 'https://scan.chainflip.io';
 export const CHAIN_FLIP_SUPPORTED_MAINNET_MAPPING = {
-  [COMMON_CHAIN_SLUGS.POLKADOT]: Chains.Polkadot,
-  [COMMON_CHAIN_SLUGS.ETHEREUM]: Chains.Ethereum
+  [COMMON_CHAIN_SLUGS.POLKADOT]: 'Polkadot',
+  [COMMON_CHAIN_SLUGS.ETHEREUM]: 'Ethereum'
 };
 export const CHAIN_FLIP_SUPPORTED_TESTNET_MAPPING = {
-  [COMMON_CHAIN_SLUGS.ETHEREUM_SEPOLIA]: Chains.Ethereum,
-  [COMMON_CHAIN_SLUGS.CHAINFLIP_POLKADOT]: Chains.Polkadot
+  [COMMON_CHAIN_SLUGS.ETHEREUM_SEPOLIA]: 'Ethereum',
+  [COMMON_CHAIN_SLUGS.CHAINFLIP_POLKADOT]: 'Polkadot'
 };
 export const CHAIN_FLIP_SUPPORTED_MAINNET_ASSET_MAPPING = {
-  [COMMON_ASSETS.DOT]: Assets.DOT,
-  [COMMON_ASSETS.ETH]: Assets.ETH,
-  [COMMON_ASSETS.USDC_ETHEREUM]: Assets.USDC
+  [COMMON_ASSETS.DOT]: 'DOT',
+  [COMMON_ASSETS.ETH]: 'ETH',
+  [COMMON_ASSETS.USDC_ETHEREUM]: 'USDC'
 };
 export const CHAIN_FLIP_SUPPORTED_TESTNET_ASSET_MAPPING = {
-  [COMMON_ASSETS.PDOT]: Assets.DOT,
-  [COMMON_ASSETS.ETH_SEPOLIA]: Assets.ETH,
-  [COMMON_ASSETS.USDC_SEPOLIA]: Assets.USDC
+  [COMMON_ASSETS.PDOT]: 'DOT',
+  [COMMON_ASSETS.ETH_SEPOLIA]: 'ETH',
+  [COMMON_ASSETS.USDC_SEPOLIA]: 'USDC'
 };
 export const SWAP_QUOTE_TIMEOUT_MAP = {
   // in milliseconds
