diff --git a/node_modules/rn-qr-generator/android/build.gradle b/node_modules/rn-qr-generator/android/build.gradle
index 0f6b87c..7830237 100644
--- a/node_modules/rn-qr-generator/android/build.gradle
+++ b/node_modules/rn-qr-generator/android/build.gradle
@@ -4,7 +4,7 @@ def safeExtGet(prop, fallback) {
 
 buildscript {
     repositories {
-        jcenter()
+        mavenCentral()
     }
 
     dependencies {
